### PR TITLE
`Quiz exercises`: Show error message for too long answer options in short answer questions

### PR DESCRIPTION
**Pull request by DanielKainz**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
#### Server
- [ ] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [ ] I implemented the changes with a good performance and prevented too many database calls.
- [ ] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [ ] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [ ] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

The unticked steps are not relevant for this PR.

### Motivation and Context

If fixes the Issue #4177 
The problem was that a short answer question does not accept >= 250 characters (for reference this is the same is the for the title), but no error message was shown.

### Description

I added a new error message which gets shown in the bottom in the `Invalid Input` field. I translated into in English and German. I added a corresponding test as well.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Exercises
4. Create a new Quiz Exercise
5. Add a `Short Answer Question`
6. Verify that the new error message is not shown if the length for all options is less than 250 characters
7. Enter >= 250 characters for at least one answer option 
8. Verify that the new error message is shown

#### Code Review
- [x] Review 1
- [x] Review 2
- [x] Review 3
#### Manual Tests
- [x] Test 1
- [x] Test 2
- [x] Test 3
- [x] Test 4
- [x] Test 5

### Screenshots

Error Message

### CODE REVIEW

**Review by lschlesinger — DISMISSED** [review]
I tested on TS1. I can confirm that it behaved as described. Seems to be fixed.

**Review by Olimbia23 — DISMISSED** [review]
Thank you for applying the suggestions @DanielKainz 
Code changes look good.
Approve 👍 

Ps. left two minor nit comments.

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts [hunk 1]
```diff
@@ -964,6 +968,12 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, Component
                         translateValues: { index: index + 1 },
                     });
                 }
+                if (!(shortAnswerQuestion.solutions?.filter((solution) => solution.text!.trim().length >= this.maxLength).length === 0)) {
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts:971]:
```suggestion
                if (shortAnswerQuestion.solutions?.filter((solution) => solution.text!.trim().length >= this.maxLength).length !== 0) {
```
[nit] I think this makes the code a bit more readable.

File: src/main/webapp/i18n/de/quizExercise.json [hunk 2]
```diff
@@ -175,6 +175,7 @@
                 "shortAnswerQuestionDuplicateMapping": "Frage {{index}}: Es gibt mindestens ein Mapping das doppelt ist.",
                 "questionTitleLength": "Frage {{index}}: Die Frage ist länger als 250 Zeichen. Bitte kürzen!",
                 "quizTitleLength": "Quiz: Der Titel ist länger als 250 Zeichen. Bitte kürzen!",
+                "quizAnswerOptionLength": "Frage {{index}}: Es gibt mindestens eine Antwortmöglichkeit mit mehr als 250 Zeichen. Bitte kürzen!",
```
[the remark this answers is not in the document]
Comment (DanielKainz) [reply src/main/webapp/i18n/de/quizExercise.json]: Yes, good point thanks. I changed it.

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts (hunk 1 above)
Comment (DanielKainz) [reply src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts]: That's a good point. I adapted it.

**Review by lschlesinger — APPROVED** [review]
I tested the PR on TS1 according to the testing steps. The error message was shown when the option length exceeded 250 characters and did not show otherwise. 👍

**Review by valeriebucher — APPROVED** [review]
Tested on ts1 and tried different lengths greater and less than 250. The error message was shown correctly

**Review by Strohgelaender — APPROVED** [review]
Tested on ts2, worked as expected. Code changes look good to me.

**Review by lBruchmann — APPROVED** [review]
Tested on ts2. I followed the testing steps. Everything worked as expected. As expected, short answers cannot be longer than 250 characters and the corresponding error message is shown.

**Pull request merged**